test/APIBasedAuth.test.ts

```diff
@@ -98,12 +98,12 @@ describe('with auth successfully created', () => {
 
   test('confirmPasswordlessAuth throws error on failed post', async () => {
     clientAxios.post = jest.fn((_path: string) => {
-      throw new Error('test error');
+      return Promise.reject('test error');
```

Comment:
Can you add a lint rule so that we don't accidentally start throwing non-Error objects. This should have been caught by a lint rule. @virdesai